### PR TITLE
Update test configuration and make paths compatible with Windows

### PR DESCRIPTION
**Pull request by namurphy**
We've been getting a persistent Python 3.9 test failure not because of anything related to xrtpy, but rather because we had a problem with the test workflow itself.  This PR updates the GitHub Action for running tests in an attempt to resolve that recurring test failure.

### CODE REVIEW

**Comment by namurphy** [comment]
pre-commit.ci autofix

File: xrtpy/response/channel.py [hunk 1]
```diff
@@ -11,16 +11,14 @@
...
 import sunpy.io.special
 import sunpy.time
 
 from astropy import units as u
+from pathlib import Path
 
-filename = pkg_resources.resource_filename(
-    "xrtpy", "response/data/xrt_channels_v0016.genx"
-)
+filename = Path(__file__).parent.absolute() / "data" / "xrt_channels_v0016.genx"
```
Comment (namurphy) [inline xrtpy/response/channel.py:21]: I ran into some issues on Windows with paths not being recognized, so I ended up using `pathlib` instead which I learned recently is the recommended way to deal with paths in Python nowadays.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -5,43 +5,32 @@ on:
...
+    - v*.*.x
   pull_request:
   workflow_dispatch:
 
 jobs:
 
   initial-tests:
     name: ${{ matrix.name }}
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```
Comment (namurphy) [inline .github/workflows/testing.yml:16]: The old version of this line meant that the tests were only run on Linux, and not on mac or Windows.  Oops!

File: xrtpy/response/effective_area.py [hunk 3]
```diff
@@ -106,7 +105,7 @@ def ccd_observation_date_to_seconds(self):
             t0 = _ccd_contamination_file_time[0]
             dt = time - t0
             ccd_observation_date_to_seconds.append(
-                (self.observation_date + timedelta(0, dt)).strftime("%s")
+                (self.observation_date + timedelta(0, dt)).strftime("%S")
```
Comment (namurphy) [inline xrtpy/response/effective_area.py:108]: For some reason, the lowercase `%s` seemed to be working fine on Linux but not on Windows.  Not sure why!

File: xrtpy/response/channel.py (hunk 1 above)
Comment (joyvelasquez) [reply xrtpy/response/channel.py:21]: Awesome, thanks for the note!

File: .github/workflows/testing.yml (hunk 2 above)
Comment (joyvelasquez) [reply .github/workflows/testing.yml:16]: Oh, understood!

File: xrtpy/response/effective_area.py (hunk 3 above)
Comment (joyvelasquez) [reply xrtpy/response/effective_area.py:108]: The (s)mall details.... Thanks for the update to the code Nick!

**Pull request merged**